modular_skyrat/modules/medical/code/anesthetic_machine.dm

```diff
@@ -0,0 +1,191 @@
+
+	if(mask_out)
+		add_overlay("mask_off")
+		return
+	add_overlay("mask_on")
+
+/obj/machinery/anesthetic_machine/attack_hand(mob/living/user)
+	. = ..()
+	if(!retract_mask())
+		return FALSE
+	visible_message(span_notice("[user] retracts [attached_mask] back into [src]."))
+
+/obj/machinery/anesthetic_machine/attacked_by(obj/item/used_item, mob/living/user)
+	if(!istype(used_item, /obj/item/tank))
+		return ..()
+
+	if(attached_tank) // If there is an attached tank, remove it and drop it on the floor
+		attached_tank.forceMove(loc)
+
+	used_item.forceMove(src) // Put new tank in, set it as attached tank
+	visible_message(span_notice("[user] inserts [used_item] into [src]."))
+	attached_tank = used_item
+	update_icon()
+
+/obj/machinery/anesthetic_machine/AltClick(mob/user)
+	. = ..()
+	if(!attached_tank)
+		return
+
+	attached_tank.forceMove(loc)
+	to_chat(user, span_notice("You remove the [attached_tank]."))
+	attached_tank = null
+	update_icon()
+	if(mask_out)
+		retract_mask()
+
+/obj/machinery/anesthetic_machine/proc/retract_mask()
+	if(!mask_out)
+		return FALSE
+
+	if(iscarbon(attached_mask.loc)) // If mask is on a mob
+		var/mob/living/carbon/attached_mob = attached_mask.loc
+		attached_mob.transferItemToLoc(attached_mask, src, TRUE)
+		attached_mob.internal = null
+	else
+		attached_mask.forceMove(src)
+
+	mask_out = FALSE
+	update_icon()
+	return TRUE
+
+/obj/machinery/anesthetic_machine/MouseDrop(mob/living/carbon/target)
+	. = ..()
+	if(!iscarbon(target))
+		return
+
+	if((!Adjacent(target)) || !(usr.Adjacent(target)))
+		return FALSE
+
+	if(!attached_tank || mask_out)
+		to_chat(usr, span_warning("[mask_out ? "The machine is already in use!" : "The machine has no attached tank!"]"))
+		return FALSE
+
+	usr.visible_message(span_warning("[usr] attemps to attach the [attached_mask] to [target]."), span_notice("You attempt to attach the [attached_mask] to [target]"))
+	if(!do_after(usr, 5 SECONDS, target))
+		return
+	if(!target.equip_to_appropriate_slot(attached_mask))
+		to_chat(usr, span_warning("You are unable to attach the [attached_mask] to [target]!"))
+		return
+
+	usr.visible_message(span_warning("[usr] attaches the [attached_mask] to [target]."), span_notice("You attach the [attached_mask] to [target]"))
+
+	target.internal = attached_tank
+	mask_out = TRUE
+	START_PROCESSING(SSmachines, src)
+	target.update_internals_hud_icon(1)
+	update_icon()
+
+/obj/machinery/anesthetic_machine/process()
+	if(!mask_out) // If not on someone, stop processing
+		return PROCESS_KILL
+
+	if(get_dist(src, get_turf(attached_mask)) > 1) // If too far away, detach
+		to_chat(attached_mask.loc, span_warning("[attached_mask] is ripped off of your face!"))
+		retract_mask()
+		return PROCESS_KILL
+
+/obj/machinery/anesthetic_machine/Destroy()
+	if(mask_out)
+		retract_mask()
+
+	QDEL_NULL(attached_mask)
+	. = ..()
+
+/// This a special version of the breath mask used for the anesthetic machine.
+/obj/item/clothing/mask/breath/anesthetic
```

Comment:
Isn't there already a special breathing mask for the anesthetics stuff?

Reply:
yes, but it mutes the patient and it doesn't have the special flag that lets it be used at a range